components/plugins/xspl.js

```
@@ -0,0 +1,58 @@
+export function convertIframeToImagePlaceholder(htmlDoc, host) {
+	const getOriginalHtmlDoc = (iframe, decode) => {
+		let html = iframe.getAttribute('data-original-html');
+		if (!html) return;
+
+		if (decode) html = tinymce.html.Entities.decode(html);
+		return new DOMParser().parseFromString(html, 'text/html');
+	};
+
+	const convertNestedIframes = (iframe) => {
+		// Decoding has already been completed in the outer iframe processing code
+		const tempDoc = getOriginalHtmlDoc(iframe, false);
+		if (!tempDoc || !tempDoc.body) return;
+
+		tempDoc.body.querySelectorAll('iframe.d2l-xspl-box').forEach(convertNestedIframes);
```

This recursion is gross imo, but we do seem to support nesting these. I figure it's either nested `DOMParser`s or nested regex and I would still prefer the former.

Caveat: I wasn't able to test this kind of nesting without falling back to invalid HTML or something that TinyMCE mangled, so I don't know if it can actually happen in practice.